### DISCUSSION TITLE
Easier way to merge into existing object

### DISCUSSION THREAD

**Discussion by NightProgramming** [RebeccaStevens/deepmerge-ts#49] [Ideas]
Sometimes it's useful to not create a new merge result object, but to have an existing object, that is the target of the whole merging process. Right now, I think, I would have to do something like:

```
// We want to merge newestData into myDataObject:

const result = deepmerge(myDataObject, newestData);
Object.assign(myDataObject, result);
```

If there was a signature like `deepmerge(target, ...sources)` the performance might be better because no intermediate object (`result`) needs to be created. Also there would be less code required (no Object.assign). And if we don't want to merge into any existing object we could do `const result = deepmerge({}, myDataObject, newestData));`. Note: The return value is the same object we used as first argument. Without having it returned by deepmerge we would need a bit more code: `const result = {}; deepmerge(result, myDataObject, newestData);`

**Comment by NightProgramming** [comment]
@RebeccaStevens To me the question arises if the target should also be a source. So when having `deepmergeInto(target, source)` should the custom mergers receive values from both `target` and `source` in order to set the property value of source? I would say yes, because otherwise we would need to call `deepmergeInto(target, target, source)` if the result value of a `target` property should be based on its old value + the value of the `source` property.

So when calling `deepmergeInto(target, source)` I think the `values` parameter for custom merger should have two values (one from `target` property value and one from `source`. Then for instance we can concat the values and have the result be set as new property value in `target`.

The only downside I see is that people might expect the values to only come from the sources, but I think that problem can be solved via docs/jsdoc.

**Reply by RebeccaStevens** [reply]
I agree.
If users don't want values from `target`, they can use a custom merge version that just ignores the first value in `values`.
